### PR TITLE
HPCC-26165 Retrieve dropzone files without specifying server

### PR DESCRIPTION
**Pull request by wangkx**
A dropzone server (hostname or IP) has to be specified when
retrieving dropzone files in the existing FileSpray service
(both DropZoneFileSearch and DropZoneFiles). By this fix,
if a dropzone server is not specified, the service returns
the matched files in all of the servers within a dropzone.

Also report the dropzone server for each dropzone file.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/services/ws_fs/ws_fsService.cpp [hunk 1]
```diff
@@ -3165,7 +3163,7 @@ bool CFileSprayEx::onDropZoneFileSearch(IEspContext &context, IEspDropZoneFileSe
         bool serverFound = false;
         unsigned filesFound = 0;
         IArrayOf<IEspPhysicalFileStruct> files;
-        bool isIPAddressReq = isIPAddress(dropZoneServerReq);
+        bool isIPAddressReq = !isEmptyString(dropZoneServerReq) && isIPAddress(dropZoneServerReq);
```
Comment (afishbeck) [inline esp/services/ws_fs/ws_fsService.cpp:3166]: I think isIPAddress already checks for empty string?

Comment (wangkx) [reply esp/services/ws_fs/ws_fsService.cpp]: The duplicated string check will be removed.

File: esp/services/ws_fs/ws_fsService.cpp [hunk 2]
```diff
@@ -3357,14 +3354,36 @@ bool CFileSprayEx::getDropZoneFiles(IEspContext &context, const char* dropZone,
     return true;
 }
 
-//This method returns all dropzones and, if NetAddress and Path specified, returns filtered list of files.
+void CFileSprayEx::getServersInDropZone(const char *dropZoneName, IArrayOf<IConstTpDropZone> &dropZoneList, bool ECLWatchVisibleOnly, StringArray &serverList)
```
Comment (ghalliday) [inline esp/services/ws_fs/ws_fsService.cpp:3357]: trivial: unusual to have a variable that starts with a capital letter.

**Review by ghalliday — APPROVED** [review]
Looks reasonable to me

File: esp/services/ws_fs/ws_fsService.cpp (hunk 2 above)
Comment (richardkchapman) [reply esp/services/ws_fs/ws_fsService.cpp:3357]: putting 'is' on the front may be a simple fix...

**Comment by richardkchapman** [comment]
This should be squashed, and retargeted to 8.8.x

**Comment by wangkx** [comment]
@richardkchapman this PR has been squashed and retargeted to 8.8.x.

**Comment by richardkchapman** [comment]
There seem to be some unwanted submodule changes included after your rebase/retarget

**Comment by wangkx** [comment]
Not sure why the 'git submodule update --init --recursive' did not clean that. I may need to manually remove it.

**Comment by wangkx** [comment]
@richardkchapman the unwanted submodule changes have been removed.

**Review by afishbeck — APPROVED** [review]

**Pull request merged**